src/bin/dual-iir.rs

```diff
@@ -183,15 +187,34 @@ const APP: () = {
     #[init(spawn=[telemetry, settings_update, ethernet_link])]
     fn init(c: init::Context) -> init::LateResources {
         // Configure the microcontroller
-        let (mut stabilizer, _pounder) =
-            hardware::setup::setup(c.core, c.device);
+        let (mut stabilizer, _pounder) = hardware::setup::setup(
+            c.core,
+            c.device,
+            SAMPLE_BUFFER_SIZE,
+            1 << SAMPLE_TICKS_LOG2,
+        );
 
         let mut network = NetworkUsers::new(
             stabilizer.net.stack,
             stabilizer.net.phy,
             stabilizer.cycle_counter,
             env!("CARGO_BIN_NAME"),
             stabilizer.net.mac_address,
+            option_env!("BROKER")
+                .and_then(|data| {
+                    data.parse::<minimq::embedded_nal::IpAddr>().map_or_else(
+                        |err| {
+                            log::error!(
+                                "{:?}: Failed to parse broker IP ({:?}) - Falling back to default",
+                                err,
+                                data
+                            );
+                            None
+                        },
+                        |ip| Some(ip),
+                    )
+                })
+                .unwrap_or(DEFAULT_MQTT_BROKER.into()),
```

Comment:
What about making this a `net::parse_or_default_broker(option_env!("BROKER"))` helper function and move it together with `DEFAULT_MQTT_BROKER` into `net`? Until we have a runtime way to get or configure this every app will likely use this.

There are also two clippy complaints about the code.